pkg/ingester/ingester_test.go

```diff
@@ -2631,11 +2643,13 @@ func TestIngester_QueryStream(t *testing.T) {
 					require.NoError(t, err)
 					assert.Greater(t, receivedSeries, 0)
 				}
+
+				expectedNumSeries = numSeries // Querying native histograms is not supported when query sharding is active.
 			} else {
 				receivedSeries, err := runQueryAndSaveResponse(&client.QueryRequest{
 					StartTimestampMs: math.MinInt64,
 					EndTimestampMs:   math.MaxInt64,
-					Matchers:         []*client.LabelMatcher{{Type: client.EQUAL, Name: model.MetricNameLabel, Value: "foo"}},
+					Matchers:         []*client.LabelMatcher{{Type: client.NOT_EQUAL, Name: model.MetricNameLabel, Value: "bar"}}, // Should return all series.
```

Comment:
Original filter should be ok, if we made the new series by adding a new label. See my comment above.

```suggestion
					Matchers:         []*client.LabelMatcher{{Type: client.EQUAL, Name: model.MetricNameLabel, Value: "foo"}},
```